Lib/unittest/test/testmock/testmock.py

```diff
@@ -1769,6 +1769,35 @@ def test_attribute_deletion(self):
             self.assertRaises(AttributeError, getattr, mock, 'f')
 
 
+    def test_mock_does_not_raise_on_repeated_attribute_deletion(self):
+        # bpo-20239: Assigning and deleting twice an attribute raises.
+        for mock in (Mock(), MagicMock(), NonCallableMagicMock(),
+                     NonCallableMock()):
+            mock.foo = 3
+            self.assertTrue(hasattr(mock, 'foo'))
+            self.assertEqual(mock.foo, 3)
+
+            del mock.foo
+            self.assertFalse(hasattr(mock, 'foo'))
+            self.assertRaises(AttributeError, getattr, mock, 'foo')
+
+            mock.foo = 4
+            self.assertTrue(hasattr(mock, 'foo'))
+            self.assertEqual(mock.foo, 4)
+
+            del mock.foo
+            self.assertFalse(hasattr(mock, 'foo'))
+            self.assertRaises(AttributeError, getattr, mock, 'foo')
+
+
+    def test_mock_raises_when_deleting_inexistent_attribute(self):
+        for mock in (Mock(), MagicMock(), NonCallableMagicMock(),
+                     NonCallableMock()):
+            with self.assertRaises(AttributeError):
+                del mock.foo
+                del mock.foo
```

Comment:
Which of these two statements is doing the raising? Only one of them should be wrapped with the `assertRaises`.